p2panda-encryption/src/traits/key_manager.rs

```diff
@@ -20,15 +20,18 @@ pub trait PreKeyManager {
 
     type Error: Error;
 
-    fn prekey_secret(y: &Self::State) -> &SecretKey;
+    fn prekey_secret<'a>(
+        y: &'a Self::State,
+        id: &'a PreKeyId,
+    ) -> Result<&'a SecretKey, Self::Error>;
 
     fn rotate_prekey(
         y: Self::State,
         lifetime: Lifetime,
         rng: &Rng,
     ) -> Result<Self::State, Self::Error>;
 
-    fn prekey_bundle(y: &Self::State) -> LongTermKeyBundle;
+    fn prekey_bundle(y: &Self::State) -> Result<LongTermKeyBundle, Self::Error>;
```

Comment:
Related to Point 2 we allow now an error to be returned when no valid bundle exists. This is the point where the application should generate new keys